auth/providers/azure/azure.go

```diff
@@ -79,19 +80,35 @@ type authInfo struct {
 	Issuer      string
 }
 
+var (
+	cachedAuthInfo *authInfo
+	mutex          = &sync.Mutex{}
+)
+
+// New is called per authentication request
 func New(ctx context.Context, opts Options) (auth.Interface, error) {
 	c := &Authenticator{
 		Options: opts,
 	}
-	authInfoVal, err := getAuthInfo(ctx, c.Environment, c.TenantID, c.HttpClientRetryCount, getMetadata)
-	if err != nil {
-		return nil, err
+
+	if cachedAuthInfo == nil {
+		mutex.Lock()
+		defer mutex.Unlock()
+		if cachedAuthInfo == nil {
+			// getAuthInfo gets the issuer, graph, and AAD endpoints from the Azure metadata endpoint
+			// since it won't change at all, we can cache it
+			authInfoVal, err := getAuthInfo(ctx, c.Environment, c.TenantID, c.HttpClientRetryCount, getMetadata)
```

Comment:
If getAuthInfo returns an error constantly, cache will stay nil forever. Does that then lead to throttling?

Reply:
yeah, but if it cannot really get result from it, there is no point to proceed.